src/MSALWrapper/AuthFlow/CachedAuth.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.AuthFlow
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Identity.Client;
+
+    /// <summary>
+    /// This class provides an abstraction over how to use the MSAL Cached Auth (known as GetTokenSilent).
+    /// </summary>
+    public static class CachedAuth
+    {
+        /// <summary>
+        /// Try to get a token silently.
+        /// </summary>
+        /// <param name="logger">An <see cref="ILogger"/>.</param>
+        /// <param name="cachedAuthTimeout">The timeout for getting a cached token.</param>
+        /// <param name="scopes">Auth Scopes.</param>
+        /// <param name="account"><see cref="IAccount"/> object to use or null if no account has been found.</param>
+        /// <param name="pcaWrapper">An <see cref="IPCAWrapper"/>.</param>
+        /// <param name="errors">List of errors to report any Exceptions into.</param>
+        /// <returns>A <see cref="Task{TResult}"/> or null if no cached token was acquired.</returns>
+        public static async Task<TokenResult> TryCachedAuthAsync(ILogger logger, TimeSpan cachedAuthTimeout, IEnumerable<string> scopes, IAccount account, IPCAWrapper pcaWrapper, IList<Exception> errors)

```

Comment:
Can we simplify this name to `TryAsync` or `ExecuteAsync`  as the class is already called `CachedAuth`?

Reply:
A comment after my own heart. Delightful call out.

Reply:
Brilliant suggestion! Thanks, @Haard30!

Reply:
`GetTokenAsync` for parity with other proper flows? 🤔

Reply:
Yeah, GetTokenAsync matches up nicely with where this might go soon :)